### PR TITLE
fixed some clippy warnings

### PR DESCRIPTION
**Pull request by llogiq**
This is mostly minor readability stuff, the code is already quite nice. :+1:

### CODE REVIEW

File: src/graphmap.rs [hunk 1]
```diff
@@ -451,14 +451,9 @@ impl<N, E> Default for GraphMap<N, E>
 /// in particular in the combination where the node type for
 /// `GraphMap` is something of type for example `Ptr(&Cell<T>)`,
 /// with the `Cell<T>` being `TypedArena` allocated.
+#[derive(Copy, Clone)]
```
Comment (bluss) [inline src/graphmap.rs]: This one is wrong, deriving inserts bounds on `T`. Is that a bug in clippy?

File: src/scored.rs [hunk 2]
```diff
@@ -33,22 +33,15 @@ impl<K: PartialOrd, T> Ord for MinScored<K, T> {
...
-            // these are the NaN cases
-            if a != a && b != b {
-                Ordering::Equal
-            } else if a != a {
-            // Order NaN less, so that it is last in the MinScore order
-                Ordering::Less
-            } else {
-                Ordering::Greater
-            }
+        match b.partial_cmp(a) {
```
Comment (bluss) [inline src/scored.rs]: It doesn't matter much, but I don't think this one is wanted -- unless  I misremember, calling partial_cmp on floats is pretty terrible performance wise (and the explicit operators better).

**Comment by bluss** [comment]
Thanks. There's lots of great small fixes there and some I don't care for. I don't agree with for example changing match to if let just because it's possible.

File: src/graphmap.rs (hunk 1 above)
Comment (llogiq) [inline src/graphmap.rs]: Yep, that's a false positive. Though on a fairly uncommon use case.

File: src/scored.rs (hunk 2 above)
Comment (llogiq) [inline src/scored.rs]: TIL. Interesting. I wonder why this is the case.

**Comment by llogiq** [comment]
Always good to learn something new. Feel free to pick the changes you like (of if I find the time I'll roll back the unwanted ones).

Replacing `match` with `if let` is deemed a readability oomprovememt, because a) less syntactic elements to read and b) it implies one indentation less (which in a good number of cases leads to less line wrapping / over-long lines).

File: src/graph.rs [hunk 3]
```diff
@@ -1,4 +1,5 @@
-//! `Graph<N, E, Ty, Ix>` is a graph datastructure using an adjacency list representation.
+//! `Graph<N, E, Ty, Ix>` is a graph datastructure using an adjacency list 
+//! representation.
```
Comment (bluss) [inline src/graph.rs]: no

File: src/graph.rs [hunk 4]
```diff
@@ -498,12 +497,9 @@ impl<N, E, Ty, Ix> Graph<N, E, Ty, Ix>
...
         if let Some(ix) = self.find_edge(a, b) {
-            match self.edge_weight_mut(ix) {
-                Some(ed) => {
-                    *ed = weight;
-                    return ix;
-                }
-                None => {}
+            if let Some(ed) = self.edge_weight_mut(ix) {
+                *ed = weight;
+                return ix;
```
Comment (bluss) [inline src/graph.rs]: Ok, I do come around to this being easier to read.

File: src/graph.rs [hunk 5]
```diff
@@ -546,11 +542,8 @@ impl<N, E, Ty, Ix> Graph<N, E, Ty, Ix>
     /// endpoint in the displaced node.
     pub fn remove_node(&mut self, a: NodeIndex<Ix>) -> Option<N>
     {
-        match self.nodes.get(a.index()) {
-            None => return None,
-            _ => {}
-        }
-        for d in DIRECTIONS.iter() {
+        if self.nodes.get(a.index()).is_none() { return None }
```
Comment (bluss) [inline src/graph.rs]: I'd prefer if this was not a oneliner but a regular if with line break after { and ; after return None.

File: src/graph.rs [hunk 6]
```diff
@@ -930,7 +923,7 @@ impl<N, E, Ty, Ix> Graph<N, E, Ty, Ix>
     /// - `Directed`, `Incoming`: All edges to `a`.
     pub fn walk_edges_directed(&self, a: NodeIndex<Ix>, dir: EdgeDirection) -> WalkEdges<Ix>
     {
-        let first_edge = self.first_edge(a, dir).unwrap_or(EdgeIndex::end());
+        let first_edge = self.first_edge(a, dir).unwrap_or_else(EdgeIndex::end);
```
Comment (bluss) [inline src/graph.rs]: no

File: src/graphmap.rs [hunk 7]
```diff
@@ -117,7 +117,7 @@ impl<N, E> GraphMap<N, E>
 
     /// Add node `n` to the graph.
     pub fn add_node(&mut self, n: N) -> N {
-        self.nodes.entry(n).or_insert_with(|| Vec::new());
+        self.nodes.entry(n).or_insert_with(Vec::new);
```
Comment (bluss) [inline src/graphmap.rs]: no

File: src/visit.rs [hunk 8]
```diff
@@ -340,7 +340,7 @@ pub trait GetAdjacencyMatrix : Graphlike {
     fn is_adjacent(&self, matrix: &Self::AdjMatrix, a: Self::NodeId, b: Self::NodeId) -> bool;
 }
 
-/// The **GraphMap** keeps an adjacency matrix internally.
+/// The `GraphMap? keeps an adjacency matrix internally.
```
Comment (bluss) [inline src/visit.rs]: typo

**Comment by bluss** [comment]
Ok, now it has a full review. The unmentioned changes are all very good.

File: src/graph.rs (hunk 3 above)
Comment (llogiq) [inline src/graph.rs]: I blame my editor :smile:

**Comment by llogiq** [comment]
Glad you like them. :smile:

**Comment by bluss** [comment]
Thanks!

**Comment by llogiq** [comment]
You're welcome.

**Pull request merged**